xc/device.ml

```diff
@@ -1493,7 +1493,15 @@ module Dm = struct
 
     let qmp_event_handle domid qmp_event =
       (* This function will be extended to handle qmp events *)
-      debug "Got QMP event, domain-%d: %s" domid qmp_event.event
+      debug "Got QMP event, domain-%d: %s" domid qmp_event.event;
+      qmp_event.data >>= fun d ->
+      match d with
+      | RTC_CHANGE timeoffset ->
+        with_xs (fun xs ->
+          let timeoffset_key = "/vm/" ^ (Uuidm.to_string (Xenops_helpers.uuid_of_domid ~xs domid)) ^ "/rtc/timeoffset" in
+          let rtc = try xs.Xs.read timeoffset_key with Xs_protocol.Enoent _ -> "0" in
+          xs.Xs.write timeoffset_key (Int64.to_string (Int64.add (Int64.of_string rtc) timeoffset))
```

Comment:
You can open Int64 for an expression like this:
```
Int64.(add timeoffset (of_string rtc) |> to_string))
```

Can `xs.Xs.read` raise other exceptions than `Enoent`?  Should these be caught and logged?
Is it safe to use "0" as timeoffset in the case of a read error? Could using "0" lead to a race because the value is too small?

Reply:
How about this:
```
          try
            let rtc = xs.Xs.read timeoffset_key in
            xs.Xs.write timeoffset_key Int64.(add timeoffset (of_string rtc) |> to_string)
          with e -> debug "Failed to process RTC_CHANGE: %s" (Printexc.to_string e)
```

Reply:
I like this better but I think this should be at least `warn`. Given that we control communication via xenstore, it should probably be even `error`.

Reply:
Thanks, now set the `error` log level.